### ISSUE TITLE
Build an interface between DVC and this Node VS Code extension

### ISSUE THREAD

**Issue by rogermparent** [iterative/vscode-dvc#3]
Through some means, this Node-based VS Code plugin will have to interface with Python-based DVC.

Using shell commands and parsing JSON output is the simplest way to get these two talking and already has some support with `show-json` flags, so the rest of the ticket assumes this will be the strategy for now.

While ideally all DVC functionality is accessible in this interface, some features are more important than others. 
This issue will probably be closed once there is enough functionality to power this extension.

### To fulfill #1: 
 We'll need to read the data printed by `dvc exp show --show-json` and return it as an object with a TS interface.

### To fulfill #2: 
- Make a virtual tree with DVC-tracked files
  We have to find some way to get a list of DVC-tracked files onto the CLI for Node to consume.
  I'm still figuring out the inner workings of DVC, but I don't see a command that does this so I believe the solution here is to parse and derive the information from `dvc.yml`, possibly reading connected `.dvc` files for more info?

- Assign actions per DVC metafiles/DVC-tracked files- pull/push/etc
  We'll need an interface to construct DVC CLI commands in JS that covers at least baseline functionality.

- Global actions related to the data management- run dvc pull, dvc checkout, dvc push
  This is practically just adding buttons to run these commands, but we'll have to figure out how to use the output to report the status of these commands- I'm sure we can check for errors, but do we want to inform the user of any other data like the amount of files changed? There may be a relevant `--show-json` here, I haven't checked yet.

### To implement [this design for the experiments view](https://files.slack.com/files-pri/T6Z5CT96G-F01CV5ZA8F2/image.png):

Basically the requirements of #1, but with a different set of data and the addition of plots. I believe that this is the data from `dvc exp show` with a different set of custom displayed parameters from the screenshot on #1. In that case, which parameters and metrics should the extension request and display? Is there a downside to just requesting as much data as possible from DVC and letting the user toggle visibility in the UI? What pieces of data are compared in the chart, and how configurable will they need to be?

### Required changes to DVC:
- [x] --show-json on `exp show` iterative/dvc#4688
